sdk/instance/instance.go

```diff
@@ -0,0 +1,117 @@
+package instancesdk
+
+import (
+	"crypto/sha256"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/docker/docker/pkg/archive"
+	"github.com/mesg-foundation/core/container"
+	"github.com/mesg-foundation/core/database"
+	"github.com/mesg-foundation/core/service"
+	"github.com/mesg-foundation/core/service/manager"
+	"github.com/mesg-foundation/core/x/xos"
+	"github.com/mr-tron/base58"
+)
+
+// Instance exposes service instance APIs of MESG.
+type Instance struct {
+	m          manager.Manager
+	container  container.Container
+	db         database.ServiceDB
+	instanceDB database.InstanceDB
+}
+
+// New creates a new Instance SDK with given options.
+func New(m manager.Manager, c container.Container, db database.ServiceDB, instanceDB database.InstanceDB) *Instance {
+	return &Instance{
+		m:          m,
+		container:  c,
+		db:         db,
+		instanceDB: instanceDB,
+	}
+}
+
+// Create creates a new service instance for service with id(sid/hash) and applies given env vars.
+func (i *Instance) Create(id string, env []string) (*service.Instance, error) {
```

Comment:
```suggestion
func (i *Instance) Create(serviceHash string, env []string) (*service.Instance, error) {
```
Not blocking but I would rename the id variable to something way more explicit.

Reply:
Discussed with @antho1404, I'll change the name but still resolve the sid inside to service hash when sid is given in the argument.